Game/Source/GDKTestGyms/GDKTestGymsGameInstance.cpp

```diff
@@ -21,19 +21,42 @@ void UGDKTestGymsGameInstance::OnStart()
 	}
 }
 
-bool UGDKTestGymsGameInstance::Tick(float DeltaSeconds)
+float UGDKTestGymsGameInstance::AddAndCalcFps(int64 NowReal, float DeltaS)
 {
-	float Alpha = 0.8f;
-	float CurrFPS = 1.0f / DeltaSeconds; // TODO - we should remove FApp::GetIdleTime() from DeltaSeconds once the NFR regex is fixed
-	AverageFPS = (Alpha * AverageFPS) + ((1.0f - Alpha) * CurrFPS);
+	TicksForFPS.Push(TPair<int64, float>(NowReal, DeltaS)); 
+	float TotalDelta = 0.0f;
+	int NumSamples = 0;
+	int64 MinRange = NowReal - FTimespan::FromMinutes(2.0).GetTicks();
+	for (int i = TicksForFPS.Num()-1; i >= 0; i--)
+	{
+		const FPSTimePoint& TimePoint = TicksForFPS[i];
+		if (TimePoint.Key < MinRange)
+		{
+			TicksForFPS.RemoveAt(i);
+		}
+		else
+		{
+			TotalDelta += TimePoint.Value;
+			NumSamples++;
+		}
+	}
+	if (NumSamples > 0)
+	{
+		return static_cast<float>(NumSamples) / TotalDelta; // 1.0f / Total / Samples 
+	}
+	return 60.0f; // If there are no samples return the ideal
```

Comment:
This isn't possible right?

Reply:
Probably not, unless the delta passed in fails the 2 minute check